presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2802,6 +2802,306 @@ public void testCollectColumnStatisticsOnInsert()
+                        "('c_varchar', null, null, null, null, null, null), " +
+                        "('c_varbinary', null, null, null, null, null, null), " +
+                        "('p_varchar', 24.0E0, 6.0E0, 0.25E0, null, null, null), " +
+                        "(null, null, null, null, 16.0E0, null, null)");
+
+        // Run analyze on 3 partitions including the null partition
+        assertUpdate(format("ANALYZE  %s WITH (partitions = ARRAY[ARRAY['p1'], ARRAY['p2'], ARRAY[NULL]])", tableName), 12);
+
+        assertQuery(format("SHOW STATS FOR (SELECT * FROM %s WHERE p_varchar = 'p1')", tableName),
+                "SELECT * FROM VALUES " +
+                        "('c_boolean', null, 2.0E0, 0.5E0, null, null, null), " +
+                        "('c_bigint', null, 2.0E0, 0.5E0, null, '0', '1'), " +
+                        "('c_double', null, 2.0E0, 0.5E0, null, '1.2', '2.2'), " +
+                        "('c_timestamp', null, 2.0E0, 0.5E0, null, '2012-08-08 00:00:00.000', '2012-08-08 01:00:00.000'), " +
+                        "('c_varchar', 8.0E0, 2.0E0, 0.5E0, null, null, null), " +
+                        "('c_varbinary', 8.0E0, null, 0.5E0, null, null, null), " +
+                        "('p_varchar', 8.0E0, 1.0E0, 0.0E0, null, null, null), " +
+                        "(null, null, null, null, 4.0E0, null, null)");
+        assertQuery(format("SHOW STATS FOR (SELECT * FROM %s WHERE p_varchar = 'p2')", tableName),
+                "SELECT * FROM VALUES " +
+                        "('c_boolean', null, 2.0E0, 0.5E0, null, null, null), " +
+                        "('c_bigint', null, 2.0E0, 0.5E0, null, '1', '2'), " +
+                        "('c_double', null, 2.0E0, 0.5E0, null, '2.3', '3.3'), " +
+                        "('c_timestamp', null, 2.0E0, 0.5E0, null, '2012-09-09 00:00:00.000', '2012-09-09 01:00:00.000'), " +
+                        "('c_varchar', 8.0E0, 2.0E0, 0.5E0, null, null, null), " +
+                        "('c_varbinary', 8.0E0, null, 0.5E0, null, null, null), " +
+                        "('p_varchar', 8.0E0, 1.0E0, 0.0E0, null, null, null), " +
+                        "(null, null, null, null, 4.0E0, null, null)");
+        assertQuery(format("SHOW STATS FOR (SELECT * FROM %s WHERE p_varchar IS NULL)", tableName),
+                "SELECT * FROM VALUES " +
+                        "('c_boolean', null, 1.0E0, 0.0E0, null, null, null), " +
+                        "('c_bigint', null, 4.0E0, 0.0E0, null, '4', '7'), " +
+                        "('c_double', null, 4.0E0, 0.0E0, null, '4.7', '7.7'), " +
+                        "('c_timestamp', null, 4.0E0, 0.0E0, null, '1977-07-07 07:04:00.000', '1977-07-07 07:07:00.000'), " +
+                        "('c_varchar', 16.0E0, 4.0E0, 0.0E0, null, null, null), " +
+                        "('c_varbinary', 16.0E0, null, 0.0E0, null, null, null), " +
+                        "('p_varchar', 0.0E0, 1.0E0, 1.0E0, null, null, null), " +
```

Comment:
Shouldn't this be `"('p_varchar', 0.0E0, 0.0E0, 1.0E0, null, null, null), " ? i.e NDV for the NULL partitioning key should be 0. 
(Also, you test that NDV does not include NULL values in `testAnalyzeUnpartitionedTable`)

Reply:
It will be fixed with https://github.com/prestodb/presto/pull/11463

Reply:
👍 
as  #11463 is merged, be sure to update the code here.